InMobiAdapter/src/main/java/com/chartboost/mediation/inmobiadapter/InMobiAdapter.kt

```diff
@@ -398,20 +409,32 @@ class InMobiAdapter : PartnerAdapter {
                         return Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_INVALID_BANNER_SIZE))
                     }
 
-                    return suspendCoroutine { continuation ->
+                    return suspendCancellableCoroutine { continuation ->
                         // Load an InMobiBanner
-                        InMobiBanner(activity, placement).apply {
+                        inMobiBannerAds[request.identifier] = InMobiBanner(activity, placement)
+                        inMobiBannerAds[request.identifier]?.apply {
                             setEnableAutoRefresh(false)
                             setBannerSize(size.width, size.height)
-                            setListener(
-                                buildBannerAdListener(
-                                    request = request,
-                                    partnerAdListener = partnerAdListener,
-                                    continuation = continuation
+                            setListener(buildBannerAdListener(
```

Comment:
this listener here, the result of buildBannerAdListener is likely the one getting garbage collected. nothing is holding onto this, right?

Reply:
It was also my original thought. Although, during the various attempts I tried, I kept seeing the same issue. 🤷🏼

Reply:
ooooooh I see. yeah, nothing is holding onto the original ad. You might be able to solve this by passing in the created banner ad to buildBannerAdListener and using that to store the PartnerAd.ad instead of the returned one.
... but also.... if this does work, then I guess it's also fine?

Reply:
Tried it, but didn't have luck. I pushed a change that still works based on your feedback. Will try again at passing the banner ad to the `buildBannerAdListener`. Just to double-check.

Reply:
Ah. Yes, for banners, the passing of the banner ad to the `buildBannerAdListener` works. Yeah, only on banners though. Pushed a commit.👍🏼